os/kernel/binary_manager/binary_manager_getinfo.c

```diff
@@ -98,6 +102,65 @@ int binary_manager_get_index_with_name(char *bin_name)
 	return ERROR;
 }
 
+/****************************************************************************
+ * Name: binary_manager_get_available_size
+ *
+ * Description:
+ *	 This function gets available size in file system.
+ *
+ ****************************************************************************/
+int binary_manager_get_available_size(int bin_idx)
+{
+	DIR *dirp;
+	int size;
+	int name_len;
+	char *bin_name;
+	struct stat file_buf;
+	struct statfs fs_buf;
+	char filepath[BIN_FILEPATH_MAX];
+	char running_file[BIN_FILENAME_MAX];
+
+	snprintf(running_file, BIN_FILENAME_MAX, "%s_%s", BIN_NAME(bin_idx), BIN_VER(bin_idx));
+	bin_name = BIN_NAME(bin_idx);
+	name_len = strlen(bin_name);
+
+	/* Get available size on file system */
+	if (statfs(BINARY_MNT_PATH, &fs_buf) != OK) {
+		bmdbg("Failed to stat fs, %s\n", BINARY_MNT_PATH);
+		return BINMGR_OPERATION_FAIL;
+	}
+	size = fs_buf.f_bavail * 1024;
```

Comment:
f_bavail means that free block.
So size is depend on block size, it is changeable value, not fixed as 1024.